modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/DeleteCTCommentMVCResourceCommand.java

```diff
@@ -33,13 +43,56 @@ protected void doServeResource(
 			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
 		throws Exception {
 
+		ThemeDisplay themeDisplay = (ThemeDisplay)resourceRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
 		long ctCommentId = ParamUtil.getLong(resourceRequest, "ctCommentId");
 
+		CTComment ctComment = ctCommentLocalService.getCTComment(ctCommentId);
+
+		Role publicationAdministratorRole = _roleLocalService.fetchRole(
+			themeDisplay.getCompanyId(), PublicationRoleConstants.NAME_ADMIN);
+
+		if (!_hasAdminRole(
+				themeDisplay.getPermissionChecker(),
+				publicationAdministratorRole) &&
+			(ctComment.getUserId() != themeDisplay.getUserId())) {
+
+			JSONPortletResponseUtil.writeJSON(
+				resourceRequest, resourceResponse,
+				JSONUtil.put(
+					"errorMessage",
+					_language.get(
+						themeDisplay.getLocale(),
+						"you-do-not-have-permission-to-delete-this-item")));
+
+			return;
+		}
+
 		ctCommentLocalService.deleteCTComment(ctCommentId);
 
 		JSONPortletResponseUtil.writeJSON(
 			resourceRequest, resourceResponse,
 			getCTCommentsJSONObject(resourceRequest));
 	}
 
+	private boolean _hasAdminRole(
+		PermissionChecker permissionChecker,
+		Role publicationAdministratorRole) {
+
+		if (permissionChecker.isCompanyAdmin() ||
+			(publicationAdministratorRole != null)) {
```

Comment:
(publicationAdministratorRole != null) only checks if the role exists in the system. We still need to check if the user has this role.